python/lsst/sims/catUtils/utils/LightCurveGenerator.py

```diff
@@ -333,15 +333,28 @@ def _get_query_from_group(self, grp, chunk_size, lc_per_field=None):
 
         lc_per_field specifies the maximum number of light curves to return
         per field of view (None implies no constraint).
+
+        constraint is a string containing a SQL constraint to be applied to
+        all database queries associated with generating these light curves
+        (optional).
         """
 
+        if self._constraint is not None and constraint is not None:
+            master_constraint = self._constraint + ' AND ' + constraint
+        elif self._constraint is not None and constraint is None:
+            master_constraint = self._constraint
+        elif self._constraint is None and constraint is not None:
+            master_constraint = constraint
+        else:
+            master_constraint = None
+
```

Comment:
I'm going to leave the `if/elif` block, as I think that will be easier to parse for anyone who has to fix this code in the future.